keps/sig-multicluster/5313-placement-decision-api/README.md

```diff
@@ -0,0 +1,537 @@
+  The value of required label `multicluster.x-k8s.io/placement` should not change and treated as immutable.
+  If order changes, update decision-index values accordingly so consumers can detect the new order.
+
+  If heavy churn is a concern, a scheduler may treat `decisions` as an unordered set and
+  maintain it in a deterministic order (ie, alphabetical sorting).
+  When the cluster set itself has not changed, this stable ordering produces an identical set of clusters,
+  so the API server skips the write and no extra change events reach consumers.
+
+- **Delete**: When a placement is no longer required,
+  the scheduler deletes every related `PlacementDecision` slice.
+  Consumers should react to the delete event and remove any workload previously applied to the listed clusters.
+
+### Ownership
+
+- The scheduler that creates the `PlacementDecision` owns the object.
+  It is solely responsible for all writes (`create`, `update`, `patch`, `delete`).
+  The consumers of the `PlacementDecision` MUST treat the object as read only (`get`, `list`, `watch`).
+- RBAC will enforce this contract by granting the scheduler write verbs on `PlacementDecisions`,
+  while limiting consumers to read only access.
+
+### Relationship to other SIG-Multicluster (SIG-MC) APIs
+* **ClusterProfile** The inventory. Each decision must reference a matching name `ClusterProfile`
+* **Work API** The workload. A consumer may read `PlacementDecision` then for each cluster creates `Work`.
+
+### Consumer Feedback
+Consumer feedback is intentionally out of scope for PlacementDecision.
+The PlacementDecision object's sole purpose is to publish the scheduler’s chosen cluster list.
+Once it has been created then it should be treated as read only by consumers.
+
+Allowing consumers to update the same PlacementDecision would complicate lifecycle ownership
+(whether the scheduler or the consumer is responsible for adding, updating, or removing cluster entries).
+It would also complicate the security/permission because maybe a malicious consumer could
+update the decision and move some workloads to unintended clusters.
+
+When consumers need to do feedback to the scheduler,
+they should do so through a separate channel like events, metrics,
+or maybe even a purpose built PlacementFeedback API
+so they have clear write authority and the scheduler can decide what to do with that feedback in the end.
+
+### User Stories (Optional)
+
+#### Story 1: GPU-aware AI training
+
+* ML scheduler scores every `ClusterProfile` by available GPUs, cost, etc.  
+* Scheduler writes `PlacementDecision` listing the best clusters.
+* A GitOps/Work-API syncer watches the decision and deploys the training job only to the listed clusters.
+* If cost rises or GPUs becoming unavailable, the scheduler updates `PlacementDecision` with new list,
+  the syncer act accordingly.
+
+#### Story 2: Progressive rollout
+
+* A progressive rollout controller begins with some canary clusters.  
```

Comment:
I wonder when is the schedule decision made? Is it made before the starting of the rollout or on the fly? The system I've used mostly have the decision made before the rollout while the rollout component controls the progress.